runtime/libia2/include/ia2_threads.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
```

Comment:
If this is only intended to be used in ia2 sources then maybe let's put it in runtime/libia2.

Reply:
If this is difficult because permissive_mode.h is a header, it's fine to leave it as-is btw.

Reply:
> If this is difficult because permissive_mode.h is a header, it's fine to leave it as-is btw.

Yeah, I need to include it in `permissive_mode.h` so it's difficult to do this.